### DISCUSSION TITLE
Emulate browser's bfcache when using client side navigation

### DISCUSSION THREAD

**Discussion by devxpy** [remix-run/remix#6636] [Proposals]
When navigating back and forward, the browser will cache the history stack - https://web.dev/bfcache/

This makes a huge difference in performance -

https://github.com/remix-run/remix/assets/19492893/ba35e4e2-bf72-43c2-9ac1-e1f36da989bc

On iOS safari, gesture navigation results in terrible UX -

https://github.com/remix-run/remix/assets/19492893/9873af51-1f6a-4db3-bb15-8da9ce40b5d9

Is there any way remix can emulate this behaviour when using client-side navigation via `<Link>`s?

**Comment by devxpy** [comment]
cc a relevant discussion from discord - https://discord.com/channels/770287896669978684/770287896669978687/1009126322162901103

**Comment by RafalFilipek** [comment]
BFCache is quite controversial. Some people really want it to work in their application, while others find it disruptive. Currently, the topic of BFCache is heavily discussed in the context of RouteCache in NextJS.

Next has added a cache for back/forward navigation in the new AppRouter, which causes major problems for applications whose state is controlled by the backend ([Deep Dive: Caching and Revalidating · vercel/next.js · Discussion #54075 (github.com)](https://github.com/vercel/next.js/discussions/54075)).

Personally, I believe that such behavior can be added, but it should be 100% possible to "opt-in" or "opt-out".

For me, for example, the new behavior is a total blocker in the context of migrating from Page Router to App Router.
